tools/cryptogenotyper/cryptogenotyper.xml

```diff
@@ -14,66 +14,56 @@
 
      #set $ref_file=''
 
-     #if $reference.ref == "no"
-        ln -s "${reference.db}" "${reference.db.name}" &&
-        #set $ref_file = $reference.db.name
+     #if $db
+        ln -s '${db}' '${db.name}' &&
+        #set $ref_file = $db.name
      #end if
 
-     #if $primers["seqtype"] == "contig"
-        ln -s "${$primers.abi_input["forward"]}" "${primers.abi_input.name}_forward.ab1" &&
-        ln -s "${$primers.abi_input["reverse"]}" "${primers.abi_input.name}_reverse.ab1" &&
-        cryptogenotyper -i "." -m "$marker" -t "$primers.seqtype" -f "forward" -r "reverse"
-        #if $reference.ref == "no"
+     #if $primers['seqtype'] == 'contig'
+        ln -s '${$primers.abi_input['forward']}' '${primers.abi_input.name}_forward.ab1' &&
+        ln -s '${$primers.abi_input['reverse']}' '${primers.abi_input.name}_reverse.ab1' &&
+        cryptogenotyper -i '.' -m '$marker' -t '$primers.seqtype' -f 'forward' -r 'reverse'
+        #if $db
           --databasefile $ref_file
         #end if
      #else
-        ln -s "${primers.abi_input}" "${primers.abi_input.element_identifier}" &&
-        cryptogenotyper -i "./${primers.abi_input.element_identifier}" -m "$marker" -t "$primers.seqtype" -f "" -r ""
-        #if $reference.ref == "no"
+        ln -s '${primers.abi_input}' '${primers.abi_input.element_identifier}' &&
+        cryptogenotyper -i './${primers.abi_input.element_identifier}' -m '$marker' -t '$primers.seqtype'
+        #if $db
           --databasefile $ref_file
         #end if
      #end if
-     #if $outputheader == "no"
+     #if $outputheader
```

Comment:
If you use the boolean from below you can remove the entire IF by simply `$outputheader` ... this works because in the true case it s "--noheaderlin" and in the no case it is an empty string.